src/lazy-image.component.ts

```diff
@@ -112,21 +115,27 @@ export class ImageComponent implements AfterViewInit, OnInit {
         const matched: ImageSource[] = this.sources.filter((source: ImageSource, _index: number, _array: ImageSource[]): boolean =>
             this.windowRef.nativeWindow.matchMedia(source.media).matches);
 
-        return matched.length > 0 ? matched[0].url : undefined;
+        return matched.length > 0 ? matched[0].url : '';
```

Comment:
this is fixing a problem with download of `undefined`

but a line above is bug when the media is not provided it will not return an image... with simple change to:
`this.windowRef.nativeWindow.matchMedia(source.media || '').matches);` <- fallback to empty string works as media all